pkg/skaffold/build/sequence.go

```diff
@@ -29,32 +29,70 @@ import (
 )
 
 // InSequence builds a list of artifacts in sequence.
-func InSequence(ctx context.Context, out io.Writer, tags tag.ImageTags, artifacts []*latest.Artifact, buildArtifact artifactBuilder) ([]Artifact, error) {
-	var builds []Artifact
+func InSequence(ctx context.Context, out io.Writer, tags tag.ImageTags, artifacts []*latest.Artifact, buildArtifact artifactBuilder) ([]chan Result, error) {
+	resultChans := make([]chan Result, len(artifacts))
+	callbackChans := make([]chan bool, len(artifacts))
 
-	for _, artifact := range artifacts {
-		color.Default.Fprintf(out, "Building [%s]...\n", artifact.ImageName)
+	for i, a := range artifacts {
+		resultChan := make(chan Result, 1)
+		resultChans[i] = resultChan
 
-		event.BuildInProgress(artifact.ImageName)
+		// when the current build is done, it will send a signal on this channel.
+		// the next build will use this channel as a start signal for its build.
+		// this way, we can chain builds together.
+		callbackChan := make(chan bool, 1)
+		callbackChans[i] = callbackChan
 
-		tag, present := tags[artifact.ImageName]
-		if !present {
-			return nil, fmt.Errorf("unable to find tag for image %s", artifact.ImageName)
+		// callback channel for the previous build. this callback tells us that the
+		// previous build is finished, so we're good to start the next one.
+		var startSignal chan bool
+		if i == 0 {
+			startSignal = make(chan bool, 1)
+			startSignal <- true // start signal for first build, since there is no previous build
+		} else {
+			startSignal = callbackChans[i-1]
 		}
+		go func(artifact *latest.Artifact, resultChan chan Result, doneChan chan bool, startSignal chan bool) {
+			<-startSignal // previous build is finished, so we can start this one
+			resultChan <- doBuild(ctx, out, tags, artifact, buildArtifact)
+			doneChan <- true
+			close(doneChan)
+			close(resultChan)
```

Comment:
"Closing is only necessary when the receiver must be told there are no more values coming, such as to terminate a range loop." We can remove this.